api_tests/applications/views/test_application_detail.py

```diff
@@ -1,35 +1,49 @@
+import pytest
 import mock
 
-from nose.tools import *  # flake8: noqa
-
 from website.models import ApiOAuth2Application, User
 from website.util import api_v2_url
-
 from tests.base import ApiTestCase
+from api_tests.utils import assert_dict_contains_subset
+from tests.json_api_test_app import JSONAPITestApp
 from osf_tests.factories import ApiOAuth2ApplicationFactory, AuthUserFactory
 
 
 def _get_application_detail_route(app):
-    path = "applications/{}/".format(app.client_id)
+    path = 'applications/{}/'.format(app.client_id)
     return api_v2_url(path, base_route='/')
 
 
 def _get_application_list_url():
-    path = "applications/"
+    path = 'applications/'
     return api_v2_url(path, base_route='/')
 
+@pytest.mark.django_db
+class TestApplicationDetail:
+
+    @pytest.fixture()
+    def app(self):
+        return JSONAPITestApp()
+
+    @pytest.fixture()
+    def user_one(self):
+        return AuthUserFactory()
 
-class TestApplicationDetail(ApiTestCase):
-    def setUp(self):
-        super(TestApplicationDetail, self).setUp()
+    @pytest.fixture()
+    def user_two(self):
```

Comment:
I would like to avoid creating a pattern of using "_one", "_two", etc. when multiple objects of a type are needed.

Instead, we can just instantiate objects in the test methods:

```python
class Test...:

    @pytest.fixture()
    def user(self):
        return AuthUserFactory()

    @pytest.fixture()
    def node(self, user):
        return NodeFactory(creator=user)

    def test_that_needs_another_user(self, node):
        non_owner = AuthUserFactory()
        res = self.app.get(..., auth=non_owner.auth)
```

This allows us to use more names that are more concise ("user") *and* more descriptive ("non_owner").

As an added bonus, this makes it much easier to refactor class-level fixture into module-level fixtures:

```python
@pytest.fixture()
def user():
    return AuthUserFactory()


class Test...:

    @pytest.fixture()
    def node(self, user):
        return NodeFactory(creator=user)

    def test_that_needs_another_user(self, node):
        non_owner = AuthUserFactory()
        res = self.app.get(..., auth=non_owner.auth)
```